crates/router/src/utils.rs

```diff
@@ -150,3 +150,7 @@ pub fn to_currency_base_unit_asf64(
     };
     Ok(amount)
 }
+
+pub fn get_payment_attempt_id(payment_id: String, attempt_count: i16) -> String {
+    format!("{payment_id}_{attempt_count}")
+}
```

Comment:
```suggestion
pub fn get_payment_attempt_id(payment_id: impl AsRef<str> + std::fmt::Display, attempt_count: i16) -> String {
    format!("{payment_id}_{attempt_count}")
}
```

This would allow you to accept any of `&str`, `String`, `&String` and anything else that satisfies those trait bounds. You won't have to always construct a `String`. Also, consider annotating with `#[inline]`.